struct initialization with type alias gives c error: struct/union type expected

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.7 18eee34.372a402
OS: windows, Microsoft Windows 11 Pro v22631 64-bit
Processor: 24 cpus, 64bit, little endian, 

getwd: D:\vbug
vexe: C:\v\v.exe
vexe mtime: 2024-09-28 05:46:02

vroot: OK, value: C:\v
VMODULES: OK, value: C:\Users\info\.vmodules
VTMP: OK, value: C:\Users\info\AppData\Local\Temp\v_0

Git version: git version 2.42.0.windows.2
Git vroot status: weekly.2024.39-33-g372a4021-dirty
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,

operable program or batch file.


thirdparty/tcc status: thirdparty-windows-amd64 b425ac82

```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg src/main.v`
```v
module main

type Mat4 = [16]f32

struct VsParams {
	mv  Mat4
	mvp Mat4
}

struct Camera {
	mv  Mat4
	mvp Mat4
}

fn main() {
	camera := Camera{}
	vs_uniforms := VsParams{
		mv:  camera.mv
		mvp: camera.mvp
	}
	println('uniforms: ${vs_uniforms}')
}

```

**What did you expect to see?**

no c error


**What did you see instead?**
```
================== C compilation error (from tcc): ==============
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01J8VRN0JP7P84KYQZG8TS62NW.tmp.c:7076: warning: implicit declaration of function 'tcc_backtrace'
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01J8VRN0JP7P84KYQZG8TS62NW.tmp.c:13544: error: struct/union type expected
... (the original output was 3 lines long, and was truncated to 2 lines)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -649,7 +649,7 @@ fn (mut g Gen) struct_init_field(sfield ast.StructInitField, language ast.Langua
 		g.inside_cast_in_heap = 0 // prevent use of pointers in child structs
 
 		field_unwrap_typ := g.unwrap_generic(sfield.typ)
-		field_unwrap_sym := g.table.sym(field_unwrap_typ)
+		field_unwrap_sym := g.table.final_sym(field_unwrap_typ)
 		if field_unwrap_sym.kind == .array_fixed && sfield.expr in [ast.Ident, ast.SelectorExpr] {
 			info := field_unwrap_sym.info as ast.ArrayFixed
 			g.fixed_array_var_init(g.expr_string(sfield.expr), sfield.expr.is_auto_deref_var(),

--- a/vlib/v/tests/structs/struct_init_alias_array_fixed_test.v
+++ b/vlib/v/tests/structs/struct_init_alias_array_fixed_test.v
@@ -1,0 +1,23 @@
+module main
+
+type Mat4 = [16]f32
+
+struct VsParams {
+	mv  Mat4
+	mvp Mat4
+}
+
+struct Camera {
+	mv  Mat4
+	mvp Mat4
+}
+
+fn test_main() {
+	camera := Camera{}
+	vs_uniforms := VsParams{
+		mv:  camera.mv
+		mvp: camera.mvp
+	}
+	assert vs_uniforms.mv.len == 16
+	assert vs_uniforms.mvp.len == 16
+}

